### PR TITLE
fix: Garabge collection of the ValidatingWebhookConfiguration

### PR DESCRIPTION
**Pull request by danielm-codefresh**
Checklist:

* [ ] My organization is added to [USERS.md](https://github.com/argoproj/argo-events/blob/master/USERS.md).

### CODE REVIEW

**Comment by danielm-codefresh** [comment]
Fixes #1602

File: manifests/install-validating-webhook.yaml [hunk 1]
```diff
@@ -58,6 +58,13 @@ rules:
   - get
   - list
   - watch
+- apiGroups:
```
Comment (whynowy) [inline manifests/install-validating-webhook.yaml:61]: Can `ClusterRole` to query `deployments` be removed?

File: webhook/cmd/start.go [hunk 2]
```diff
@@ -43,13 +43,14 @@ func Start() {
...
 	options := webhook.Options{
-		ServiceName:    "events-webhook",
-		DeploymentName: "events-webhook",
-		Namespace:      namespace,
-		Port:           443,
-		SecretName:     "events-webhook-certs",
-		WebhookName:    "webhook.argo-events.argoproj.io",
-		ClientAuth:     tls.VerifyClientCertIfGiven,
+		ServiceName:     "events-webhook",
+		DeploymentName:  "events-webhook",
```
Comment (whynowy) [inline webhook/cmd/start.go:47]: `DeploymentName` also can be removed.

Comment (danielm-codefresh) [reply webhook/cmd/start.go:47]: It is still used [here](https://github.com/argoproj/argo-events/blob/master/webhook/webhook.go#L183)

Comment (whynowy) [reply webhook/cmd/start.go:47]: Silly me...

**Review by whynowy — APPROVED** [review]
LGTM, thanks!!

**Pull request merged**